acceptance/tests/face/4654_facts_face.rb

```diff
@@ -0,0 +1,63 @@
+test_name "Puppet facts face should resolve custom and external facts"
+
+#
+# This test is intended to ensure that custom and external facts present
+# on the agent are resolved and displayed by the puppet facts face.
+#
+custom_fact = <<CFACT
+Facter.add('custom_fact') do
+  setcode do
+    'foo'
+  end
+end
+CFACT
+
+unix_external_fact = <<EFACT
+#!/bin/sh
+echo 'external_fact=bar'
+EFACT
+
+win_external_fact = <<EFACT
+@echo off
+echo external_fact=bar
+EFACT
+
+agents.each do |agent|
+  if agent['platform'] =~ /windows/
+    external_fact = win_external_fact
+    ext = '.bat'
+  else
+    external_fact = unix_external_fact
+    ext = '.sh'
+  end
+
+  step "Create a fact dir with custom and external facts"
+  factdir = agent.tmpdir('4654-factdir')
+
+  on agent, "mkdir -p '#{File.join(factdir, 'lib', 'facter')}'"
+  on agent, "mkdir -p '#{File.join(factdir, 'facts.d')}'"
+  on agent, puppet('apply'), :stdin => <<MANIFEST
+  file { "#{factdir}/lib/facter/custom.rb":
+    ensure  => file,
+    content => "#{custom_fact}",
+  }
+
+  file { "#{factdir}/facts.d/external#{ext}":
+    ensure  => file,
+    mode    => "0755",
+    content => "#{external_fact}",
+  }
+MANIFEST
+
+  step "Agent #{agent}: custom_fact and external_fact should be present in the output of `puppet facts`"
+  on agent, puppet('facts',
+                   '--libdir', File.join(factdir, 'lib'),
+                   '--pluginfactdest', File.join(factdir, 'facts.d')) do
```

Comment:
we should probably only do that if beaker is asking puppet/facter what its paths are... otherwise we probably should be defining them in options.rb, particularly if used in more than one test.